app/main/views/frameworks.py

```diff
@@ -961,23 +961,28 @@ def signature_upload(framework_slug, agreement_id):
 
     agreements_bucket = s3.S3(current_app.config['DM_AGREEMENTS_BUCKET'])
     agreement_path = agreement.get('signedAgreementPath')
-    signature_page = agreements_bucket.get_key(agreement_path) if agreement_path else None
+    existing_signature_page = agreements_bucket.get_key(agreement_path) if agreement_path else None
     upload_error = None
-
     if request.method == 'POST':
+        fresh_signature_page = request.files.get('signature_page')
+
         # No file chosen for upload and file already exists on s3 so can use existing and progress
-        if not request.files['signature_page'].filename and signature_page:
+        if not (fresh_signature_page and fresh_signature_page.filename) and existing_signature_page:
             return redirect(url_for(".contract_review", framework_slug=framework_slug, agreement_id=agreement_id))
 
-        if not file_is_image(request.files['signature_page']) and not file_is_pdf(request.files['signature_page']):
+        # Validate file
+        if not fresh_signature_page:
```

Comment:
Just to check, this change is because you spotted that we aren't covering this case currently, right?

Reply:
This is because of the new Werkzeug behaviour of (apparently) not passing empty files through to the view in the request. Previously if the button was pressed to POST to this route with no file selected then `request.files['signature_page']` existed in the view as an empty FileStorage object.  This has now changed and it strips the key out from the request altogether, so we need to deal with the case that the key doesn't exist.